libvast/src/system/accountant.cpp

```diff
@@ -64,12 +68,26 @@ void init(accountant_actor* self, const path& filename) {
 #if VAST_LOG_LEVEL >= CAF_LOG_LEVEL_INFO
   self->delayed_send(self, overview_delay, telemetry_atom::value);
 #endif
+  st.slice_size = get_or(self->system().config(), "system.table-slice-size",
+                         defaults::system::table_slice_size);
+}
+
+void finish_slice(accountant_actor* self) {
+  auto& st = self->state;
+  // Do nothing if builder has not been created.
+  if (!st.builder)
+    return;
+  auto slice = st.builder->finish();
+  VAST_DEBUG(self, "generated slice with", slice->rows(), "rows");
```

Comment:
Maybe add an assertion here that `slice->rows() > 0`?

Reply:
This was actually possible if `finish_slice()` was called from the exit handler. Rectified.